dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2PublisherDataProvider.java

```diff
@@ -0,0 +1,361 @@
+        uriBuilder.addParameter("filter", "[[\"name\",\"contains word\",\"" + query + "\"]]");
+        uriBuilder.addParameter("format", "Json");
+        // Set optional start (offset) and limit parameters
+        if (start >= 0) {
+            uriBuilder.addParameter("offset", String.valueOf(start));
+        }
+        if (limit > 0) {
+            uriBuilder.addParameter("limit", String.valueOf(limit));
+        }
+        if (StringUtils.isNotBlank(apiKey)) {
+            uriBuilder.addParameter("api-key", apiKey);
+        }
+
+        // Create HTTP GET object
+        HttpGet method = new HttpGet(uriBuilder.build());
+
+        // Set connection parameters
+        int timeout = 5000;
+        method.setConfig(RequestConfig.custom()
+            .setConnectionRequestTimeout(timeout)
+            .setConnectTimeout(timeout)
+            .setSocketTimeout(timeout)
+            .build());
+        return new HttpGet(uriBuilder.build());
+    }
+
+    private ExternalDataObject constructExternalDataObjectFromSherpaPublisher(SHERPAPublisher sherpaPublisher) {
+        ExternalDataObject externalDataObject = new ExternalDataObject();
+        externalDataObject.setSource(sourceIdentifier);
+
+        // Set publisher name
+        if (StringUtils.isNotBlank(sherpaPublisher.getName())) {
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "title", null, null, sherpaPublisher.getName()));
+            externalDataObject.setDisplayValue(sherpaPublisher.getName());
+            externalDataObject.setValue(sherpaPublisher.getName());
+        }
+        // Set publisher ID
+        if (StringUtils.isNotBlank(sherpaPublisher.getIdentifier())) {
+            externalDataObject.setId(sherpaPublisher.getIdentifier());
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "identifier", "sherpaPublisher", null,
+                sherpaPublisher.getIdentifier()));
+        }
+
+        // Set home URL
+        if (StringUtils.isNotBlank(sherpaPublisher.getUri())) {
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "identifier", "other", null, sherpaPublisher.getUri()));
+        }
+
+        return externalDataObject;
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    /**
+     * Get number of results returned from a SHERPA v2 publication search
+     * @param query The query to be search on and give the total amount of results
+     * @return int representing number of publisher results
+     */
+    @Override
+    public int getNumberOfResults(String query) {
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
```

Comment:
Can we use "HttpStatus.SC_OK" instead of 200 ?

Reply:
Done! See commit# 839159b